prov/efa/src/rxr/rxr_msg.c

```diff
@@ -172,7 +172,7 @@ int rxr_msg_select_rtm(struct rxr_ep *rxr_ep, struct rxr_tx_entry *tx_entry, int
 
 	eager_rtm_max_data_size = rxr_tx_entry_max_req_data_capacity(rxr_ep, tx_entry, eager_rtm);
 
-	if (tx_entry->total_len <= eager_rtm_max_data_size)
+	if (rxr_env.enable_eager_message && tx_entry->total_len <= eager_rtm_max_data_size)
 		return eager_rtm;
 
 	if (tx_entry->total_len <= rxr_env.efa_max_medium_msg_size)
```

Comment:
Maybe I'm not understanding the need here. If we don't want to send eager packets, why would we want to execute the medium message protocol (which is just sending a bunch of eager messages iirc).

Reply:
It is for experimental purpose. Users can further disable medium by setting the `max_medium_size` to 0